src/qt/veil/veilstatusbar.cpp

```diff
@@ -59,6 +61,39 @@ void VeilStatusBar::onBtnSyncClicked(){
 
 #ifdef ENABLE_WALLET
 bool fBlockNextStakeCheckSignal = false;
+void VeilStatusBar::setStakingText() {
+
+    // Determine if staking is recently active. Note that this is not immediate effect. Staking could be disabled and it could take some time (activeStakingMaxTime) update state.
+    int64_t nTimeLastHashing = 0;
+    if (!mapHashedBlocks.empty()) {
+        auto pindexBest = chainActive.Tip();
+        if (mapHashedBlocks.count(pindexBest->GetBlockHash())) {
+            nTimeLastHashing = mapHashedBlocks.at(pindexBest->GetBlockHash());
+        } else if (mapHashedBlocks.count(pindexBest->pprev->GetBlockHash())) {
+            nTimeLastHashing = mapHashedBlocks.at(pindexBest->pprev->GetBlockHash());
+        }
+    }
+    bool fStakingActive = false;
+    if (nTimeLastHashing)
+        fStakingActive = GetAdjustedTime() + MAX_FUTURE_BLOCK_TIME - nTimeLastHashing < ACTIVE_STAKING_MAX_TIME;
+	
+    if (syncFlag){
+	ui->checkStaking->setText("Staking Disabled while Syncing");
+    }else if (this->walletModel->isStakingEnabled()) {
+	if (fStakingActive) {
+            ui->checkStaking->setText("Staking Enabled");
+	}else{
+	    ui->checkStaking->setText("Enabling...");
```

Comment:
I think we need to checked for a locked wallet here. If the wallet is locked, display "unlock wallet for staking"
![image](https://user-images.githubusercontent.com/46700212/101856334-af42fc00-3b2a-11eb-89ca-539c29be4e0b.png)

More of NIT, when the wallet is unlocked and there are no zerocoins, the text stays at Enabling...
![image](https://user-images.githubusercontent.com/46700212/101856592-38f2c980-3b2b-11eb-89bb-fdefcc5ea11f.png)